src/main/java/br/com/zup/edu/biblioteca/validators/ValidacaoParaEmprestimoPorTipoDeUsuarioPadrao.java

```diff
@@ -0,0 +1,91 @@
+package br.com.zup.edu.biblioteca.validators;
+
+import br.com.zup.edu.biblioteca.controller.requests.CadastroEmprestimoDeExemplarRequest;
+import br.com.zup.edu.biblioteca.model.Livro;
+import br.com.zup.edu.biblioteca.model.Usuario;
+import br.com.zup.edu.biblioteca.util.ExecutorTransacional;
+import org.springframework.stereotype.Component;
+import org.springframework.validation.Errors;
+
+import javax.persistence.EntityManager;
+import javax.persistence.TypedQuery;
+import java.util.Objects;
+
+import static br.com.zup.edu.biblioteca.model.TipoCirculacao.LIVRE;
+import static br.com.zup.edu.biblioteca.model.TipoUsuario.PADRAO;
+import static javax.persistence.LockModeType.READ;
+//10
+@Component
+public  class ValidacaoParaEmprestimoPorTipoDeUsuarioPadrao implements ValidacaoEmprestimo{
+
+    //1
+    private final ExecutorTransacional execTransacional;
+
+    public ValidacaoParaEmprestimoPorTipoDeUsuarioPadrao(ExecutorTransacional execTransacional) {
+        this.execTransacional = execTransacional;
+    }
+
+
+    @Override
+    public Errors handler(Errors errors, CadastroEmprestimoDeExemplarRequest request) {
+        final EntityManager manager = execTransacional.getManager();
+        Usuario possivelResponsavelPeloEmprestimo = execTransacional.executor(()->manager.find(Usuario.class, request.getIdUsuario()));
+        //1
+        if(Objects.isNull(possivelResponsavelPeloEmprestimo)){
+            errors.rejectValue("idUsuario",null,"Usuario nao cadastrado");
+            return errors;
+        }
+
+        //1
+        Livro livroDesejado=execTransacional.executor(()-> manager.find(Livro.class,request.getIdLivro()));
+
+        if(Objects.isNull(livroDesejado)){
+            errors.rejectValue("idLivro",null,"Livro nao cadastrado");
+            return errors;
+        }
+
+        String porQuantosExemplaresEsteUsuarioEResponsavel = "SELECT COUNT(e) FROM EmprestimoDeExemplar e WHERE e.usuario=:responsavel";
+        TypedQuery<Long> query = manager.createQuery(porQuantosExemplaresEsteUsuarioEResponsavel, Long.class);
+        query.setParameter("responsavel", possivelResponsavelPeloEmprestimo);
+        Integer quantidadeDeLivrosEmprestados = execTransacional.executor(()->query.getSingleResult().intValue());
+
+        //1
+        if (possivelResponsavelPeloEmprestimo.getTipoUsuario().equals(PADRAO)) {
```

Comment:
Por que fazer esse `if` se esse validador já é para usuário padrão?

Reply:
como é disparada em cadeia as validações é necessário identificar quando algumas serão executadas. Porém a validação de tipo oposto no inicio promove melhor performance, caso seja retornado o object de Erros, quando não for do tipo padrão.